src/triage/component/catwalk/evaluation.py

```diff
@@ -110,72 +111,177 @@ def _validate_metrics(
+            short_threshold_unit = 'pct' if threshold_unit == 'percentile' else 'abs'
+            full_params[short_threshold_unit] = threshold_value
+        parameter_string = '/'.join([
+            '{}_{}'.format(val, key)
+            for key, val in full_params.items()
+        ])
+        return parameter_string
+
+    def _filter_nan_labels(self, predicted_classes, labels):
+        """Filter missing labels and their corresponding predictions
+
+        Args:
+            predicted_classes (list) Predicted binary classes, of same length as labels
+            labels (list) Labels, maybe containing NaNs
+
+        Returns: (tuple) Copies of the input lists, with NaN labels removed
+        """
+        labels = numpy.array(labels)
+        predicted_classes = numpy.array(predicted_classes)
+        nan_mask = numpy.isfinite(labels)
+        return (
+            (predicted_classes[nan_mask]).tolist(),
+            (labels[nan_mask]).tolist()
+        )
+
+    def _evaluations_for_threshold(
         self,
         metrics,
         parameters,
-        threshold_config,
         predictions_proba,
-        predictions_binary,
         labels,
+        threshold_unit,
+        threshold_value,
     ):
-        """Generate evaluations based on config and create ORM objects to hold them
+        """Generate evaluations for a given threshold in a metric group,
+        and create ORM objects to hold them
 
         Args:
             metrics (list) names of metric to compute
             parameters (list) dicts holding parameters to pass to metrics
-            threshold_config (dict) Unit type and value referring to how any
-                thresholds were computed. Combined with parameter string
-                to make a unique identifier for the parameter in the database
+            threshold_unit (string) the type of threshold, either 'percentile' or 'top_n'
+            threshold_value (int) the numeric threshold,
             predictions_proba (list) Probability predictions
-            predictions_binary (list) Binary predictions
-            labels (list) True labels
+            labels (list) True labels (may have NaNs)
 
         Returns: (list) results_schema.Evaluation objects
         Raises: UnknownMetricError if a given metric is not present in
             self.available_metrics
         """
+
+        # using threshold configuration, convert probabilities to predicted classes
+        predicted_classes = generate_binary_at_x(
+            predictions_proba,
+            threshold_value,
+            unit=threshold_unit
+        )
+        # filter out null labels
+        predicted_classes_with_labels, present_labels = self._filter_nan_labels(
+            predicted_classes,
+            labels,
+        )
+        num_labeled_examples = len(present_labels)
+        num_labeled_above_threshold = predicted_classes_with_labels.count(1)
+        num_positive_labels = present_labels.count(1)
```

Comment:
This is not a change in this PR, but does this generalize to multinomial categorization cases? If not, is that something that we desire?

Reply:
It doesn't - in general, Triage doesn't (our label generators are binary-only, for instance). Perhaps in the future we will do this, but it seems in general we split these into different binary cases anyway.